Add a new TTS recipe named `qasr_tts`

massabaali7 opened the pull request.

ftshijt reviewed (commented):
Could you please first check the list here https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#134-checklist-before-you-submit-the-recipe-based-pr ? Please use softlink instead copy for `tts.sh` and `db.sh`

massabaali7 commented:
Hello,

I updated what you've mentioned.

Best,

Massa

On Thu, 12 May 2022 at 15:11, Jiatong ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> Could you please first check the list here
> https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#134-checklist-before-you-submit-the-recipe-based-pr
> ? Please use softlink instead copy for tts.sh and db.sh
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#pullrequestreview-970670329>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDBYP24D5DZKXKCYHVVDVJTRPFANCNFSM5VUH46XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ftshijt commented on egs2/TEMPLATE/asr1/db.sh:
```diff
@@ -1,20 +1,16 @@
 # Set the path of your corpus
 # "downloads" means the corpus can be downloaded by the recipe automatically
 
-ACCENTED_FR=downloads
```
could you not remove the entries for other datasets?

massabaali7 replied:
For now, I will remove the automatic download.

kan-bayashi replied:
No, please do not remove other databases from `db.sh`.
You may restore by 
```
git fetch origin
git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh
# then add your new DB and commit it
...
```

ftshijt commented on egs2/TEMPLATE/asr1/db.sh:
```diff
@@ -47,13 +42,12 @@ LIBRILIGHT_LIMITED=
 LIBRITTS=
 LJSPEECH=downloads
-MUSAN=
+qasr_tts=downloads
```
To keep the convention of capital letter, maybe use `QASR_TTS` instead of qasr_tts

ftshijt commented on egs2/qasr_tts/RESULTS.MD:
```diff
@@ -0,0 +1,17 @@
+ 
+- Results 
+  - FastSpeech2 (Trans.) w/ PWG: (R = 1) 
+      - CER: 3.9
```
Interesting observation, can you add a note for which ASR model you are using to get the number?

massabaali7 replied:
I used one of our pre-trained ASR models trained on mgb2.

amali replied:
We can link to the MGB-2 recipe.

ftshijt commented on egs2/qasr_tts/RESULTS.MD:
```diff
@@ -0,0 +1,17 @@
+      - CER: 3.9
+      - WER: 9.13
+      - MOS
+        - intelligibility: 4.4 ± 0.06
```
It would be great if you can reveal some details for the objective evaluation (also great to have some comparisons to other models/dataset) It would be not so informative by only providing the information with single dataset.

massabaali7 replied:
I will be sharing the paper once its ready with all the trained models and comparisons.

ftshijt commented on egs2/qasr_tts/adapt.sh:
```diff
@@ -0,0 +1,31 @@
+. local/pretrained.sh
```
Cool procedure, can we have a better documentation on this script (e.g., maybe in RESULTS.md or README.md of this recipe)

massabaali7 replied:
Okay I will add a README

ftshijt commented on egs2/qasr_tts/RESULTS.MD:
```diff
@@ -0,0 +1,17 @@
+    - training config file: `./conf/tuning/train_conformer_fastspeech2.yaml`
+ 
+- Results 
+  - FastSpeech2 (Trans.) w/ PWG: (R = 1) 
```
Please also upload your model to huggingface to support the number (you can achieve it by the last stage in tts.sh)

ftshijt commented on egs2/qasr_tts/conf/tuning/finetune_joint_conformer_fastspeech2_hifigan.yaml:
```diff
@@ -0,0 +1,271 @@
+# This EXPERIMENTAL configuration is for ESPnet2 to finetune
```
For initial PR, it would be better to make the config concise. Could you try to remove some redundant config for the start?

massabaali7 replied:
Yes sure

ftshijt commented on egs2/qasr_tts/local/data.sh:
```diff
@@ -0,0 +1,87 @@
+. ./db.sh || exit 1;
+
+if [ -z "${qasr_tts}" ]; then
+   log "Fill the value of 'JSUT' of db.sh"
```
it is JSUT here, please change to your set

ftshijt commented on egs2/qasr_tts/local/pretrained.sh:
```diff
@@ -0,0 +1,2 @@
+# download pretrained model kan-bayashi/ljspeech_tts_train_transformer_raw_char_tacotron_train.loss.ave
+pretrained="https://zenodo.org/record/4925105/files/tts_train_transformer_raw_char_tacotron_train.loss.ave.zip?download=1"
```
Seems redundant, can you merge it to other script?

massabaali7 replied:
Okay

ftshijt commented on egs2/qasr_tts/run.sh:
```diff
@@ -0,0 +1,45 @@
+    --fs "${fs}" \
+    --n_fft "${n_fft}" \
+    --n_shift "${n_shift}" \
+    --token_type char \
```
you are using g2p in line 27 but use char token_type here, which could mask the effect of g2p

ftshijt commented on egs2/qasr_tts/run.sh:
```diff
@@ -0,0 +1,45 @@
+g2p=g2p_en_no_space # Include no word separator
+
+bash ./tts.sh \
+    --stage 1 \
```
```suggestion
```

ftshijt reviewed (commented):
Many thanks! Some suggestions as follows

massabaali7 commented:
Okay I added them to the tts1 directory.

On Sat, 14 May 2022 at 05:30, Tomoki Hayashi ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Could you keep the recipe naming rule?
> egs2/qasr_tts -> egs2/qasr_tts/tts1
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#pullrequestreview-972954048>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDB76WOCFRTMIIEGPUZTVJ362VANCNFSM5VUH46XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kan-bayashi commented:
@massabaali7 You use wrong path `qasr_tts/tts1` -> `egs2/qasr_tts/tts1`.

kan-bayashi commented on qasr_tts/tts1/local/data_download.sh:
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Tomoki Hayashi
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
```
Please change to yours.

kan-bayashi commented on qasr_tts/tts1/run.sh:
```diff
@@ -0,0 +1,45 @@
+train_config=conf/train.yaml
+inference_config=conf/decode.yaml
+
+# g2p=g2p_en # Include word separator
+g2p=g2p_en_no_space # Include no word separator
+
+bash ./tts.sh \
+    --stage 1 \
+    --lang ar \
+    --feats_type raw \
+    --fs "${fs}" \
+    --n_fft "${n_fft}" \
+    --n_shift "${n_shift}" \
+    --token_type char \
+    --cleaner tacotron \
+    --g2p "${g2p}" \
+    --train_config "${train_config}" \
+    --inference_config "${inference_config}" \
+    --train_set "${train_set}" \
+    --valid_set "${valid_set}" \
+    --test_sets "${test_sets}" \
+    --srctexts "data/${train_set}/text" \
+    ${opts} "$@"
```
G2P and cleaner may not be needed? (Please check)
```suggestion
./tts.sh \
    --stage 1 \
    --lang ar \
    --feats_type raw \
    --fs "${fs}" \
    --n_fft "${n_fft}" \
    --n_shift "${n_shift}" \
    --token_type char \
    --cleaner none \
    --g2p none \
    --train_config "${train_config}" \
    --inference_config "${inference_config}" \
    --train_set "${train_set}" \
    --valid_set "${valid_set}" \
    --test_sets "${test_sets}" \
    --srctexts "data/${train_set}/text" \
    ${opts} "$@"
```

kan-bayashi commented on qasr_tts/tts1/README.MD:
```diff
@@ -0,0 +1,15 @@
+- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies:
```
```suggestion
# QASR-TTS RECIPE

- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies:
```

massabaali7 commented:
I updated the repo. For the db.sh for now we will not add automatic
download. Yes, the cleaner is not needed.

On Mon, 16 May 2022 at 03:52, Tomoki Hayashi ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In qasr_tts/tts1/local/data_download.sh
> <https://github.com/espnet/espnet/pull/4357#discussion_r873247375>:
>
> > +# Copyright 2019 Tomoki Hayashi
>
> +#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
>
>
> Please change to yours.
> ------------------------------
>
> In qasr_tts/tts1/run.sh
> <https://github.com/espnet/espnet/pull/4357#discussion_r873247527>:
>
> > +# g2p=g2p_en # Include word separator
>
> +g2p=g2p_en_no_space # Include no word separator
>
> +
>
> +bash ./tts.sh \
>
> +    --stage 1 \
>
> +    --lang ar \
>
> +    --feats_type raw \
>
> +    --fs "${fs}" \
>
> +    --n_fft "${n_fft}" \
>
> +    --n_shift "${n_shift}" \
>
> +    --token_type char \
>
> +    --cleaner tacotron \
>
> +    --g2p "${g2p}" \
>
> +    --train_config "${train_config}" \
>
> +    --inference_config "${inference_config}" \
>
> +    --train_set "${train_set}" \
>
> +    --valid_set "${valid_set}" \
>
> +    --test_sets "${test_sets}" \
>
> +    --srctexts "data/${train_set}/text" \
>
> +    ${opts} "$@"
>
>
> G2P and cleaner may not be needed? (Please check)
> ⬇️ Suggested change
>
> -# g2p=g2p_en # Include word separator
>
> -g2p=g2p_en_no_space # Include no word separator
>
> -
>
> -bash ./tts.sh \
>
> -    --stage 1 \
>
> -    --lang ar \
>
> -    --feats_type raw \
>
> -    --fs "${fs}" \
>
> -    --n_fft "${n_fft}" \
>
> -    --n_shift "${n_shift}" \
>
> -    --token_type char \
>
> -    --cleaner tacotron \
>
> -    --g2p "${g2p}" \
>
> -    --train_config "${train_config}" \
>
> -    --inference_config "${inference_config}" \
>
> -    --train_set "${train_set}" \
>
> -    --valid_set "${valid_set}" \
>
> -    --test_sets "${test_sets}" \
>
> -    --srctexts "data/${train_set}/text" \
>
> -    ${opts} "$@"
>
> +./tts.sh \
>
> +    --stage 1 \
>
> +    --lang ar \
>
> +    --feats_type raw \
>
> +    --fs "${fs}" \
>
> +    --n_fft "${n_fft}" \
>
> +    --n_shift "${n_shift}" \
>
> +    --token_type char \
>
> +    --cleaner none \
>
> +    --g2p none \
>
> +    --train_config "${train_config}" \
>
> +    --inference_config "${inference_config}" \
>
> +    --train_set "${train_set}" \
>
> +    --valid_set "${valid_set}" \
>
> +    --test_sets "${test_sets}" \
>
> +    --srctexts "data/${train_set}/text" \
>
> +    ${opts} "$@"
>
>
> ------------------------------
>
> In qasr_tts/tts1/README.MD
> <https://github.com/espnet/espnet/pull/4357#discussion_r873248154>:
>
> > @@ -0,0 +1,15 @@
>
> +- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies:
>
>
> ⬇️ Suggested change
>
> -- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies:
>
> +# QASR-TTS RECIPE
>
> +
>
> +- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies:
>
>
> ------------------------------
>
> In egs2/TEMPLATE/asr1/db.sh
> <https://github.com/espnet/espnet/pull/4357#discussion_r873248083>:
>
> > @@ -1,20 +1,16 @@
>
>  # Set the path of your corpus
>
>  # "downloads" means the corpus can be downloaded by the recipe automatically
>
>
>
> -ACCENTED_FR=downloads
>
>
> No, please do not remove other databases from db.sh.
> You may restore by
>
> git fetch origin
>
> git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh
>
> # then add your new DB and commit it
>
> ...
>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#pullrequestreview-973301072>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDB5JUFFZJMEXF5JZL6DVKGE5HANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ftshijt commented:
> I updated the repo. For the db.sh for now we will not add automatic download. Yes, the cleaner is not needed.
> […](#)
> On Mon, 16 May 2022 at 03:52, Tomoki Hayashi ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In qasr_tts/tts1/local/data_download.sh <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873247375)>: > +# Copyright 2019 Tomoki Hayashi +# Apache 2.0 (http://www.apache.org/licenses/LICENSE-2.0) Please change to yours. ------------------------------ In qasr_tts/tts1/run.sh <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873247527)>: > +# g2p=g2p_en # Include word separator +g2p=g2p_en_no_space # Include no word separator + +bash ./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ + --token_type char \ + --cleaner tacotron \ + --g2p "${g2p}" \ + --train_config "${train_config}" \ + --inference_config "${inference_config}" \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts} "$@" G2P and cleaner may not be needed? (Please check) ⬇️ Suggested change -# g2p=g2p_en # Include word separator -g2p=g2p_en_no_space # Include no word separator - -bash ./tts.sh \ - --stage 1 \ - --lang ar \ - --feats_type raw \ - --fs "${fs}" \ - --n_fft "${n_fft}" \ - --n_shift "${n_shift}" \ - --token_type char \ - --cleaner tacotron \ - --g2p "${g2p}" \ - --train_config "${train_config}" \ - --inference_config "${inference_config}" \ - --train_set "${train_set}" \ - --valid_set "${valid_set}" \ - --test_sets "${test_sets}" \ - --srctexts "data/${train_set}/text" \ - ${opts} "$@" +./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ + --token_type char \ + --cleaner none \ + --g2p none \ + --train_config "${train_config}" \ + --inference_config "${inference_config}" \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts} "$@" ------------------------------ In qasr_tts/tts1/README.MD <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873248154)>: > @@ -0,0 +1,15 @@ +- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies: ⬇️ Suggested change -- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies: +# QASR-TTS RECIPE + +- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies: ------------------------------ In egs2/TEMPLATE/asr1/db.sh <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873248083)>: > @@ -1,20 +1,16 @@ # Set the path of your corpus # "downloads" means the corpus can be downloaded by the recipe automatically -ACCENTED_FR=downloads No, please do not remove other databases from db.sh. You may restore by git fetch origin git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh # then add your new DB and commit it ... — Reply to this email directly, view it on GitHub <[#4357 (review)](https://github.com/espnet/espnet/pull/4357#pullrequestreview-973301072)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYBJDB5JUFFZJMEXF5JZL6DVKGE5HANCNFSM5VUH46XA> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Hi, we are not referring to your recipe, but in the case that you remove some other entries for other recipes. You may refer the diff here https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2

massabaali7 commented:
I added the original db.sh after removing qar_tts since I prevented the
automatic download for now.

On Thu, May 19, 2022, 11:04 PM Jiatong ***@***.***> wrote:

> I updated the repo. For the db.sh for now we will not add automatic
> download. Yes, the cleaner is not needed.
> … <#m_-8517925574695521207_>
> On Mon, 16 May 2022 at 03:52, Tomoki Hayashi *@*.*> wrote: @.** commented
> on this pull request. ------------------------------ In
> qasr_tts/tts1/local/data_download.sh <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873247375>>: > +#
> Copyright 2019 Tomoki Hayashi +# Apache 2.0 (
> http://www.apache.org/licenses/LICENSE-2.0) Please change to yours.
> ------------------------------ In qasr_tts/tts1/run.sh <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873247527>>: > +#
> g2p=g2p_en # Include word separator +g2p=g2p_en_no_space # Include no word
> separator + +bash ./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw
> \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ +
> --token_type char \ + --cleaner tacotron \ + --g2p "${g2p}" \ +
> --train_config "${train_config}" \ + --inference_config "${inference_config}"
> \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ +
> --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts}
> "$@" G2P and cleaner may not be needed? (Please check) ⬇️ Suggested
> change -# g2p=g2p_en # Include word separator -g2p=g2p_en_no_space #
> Include no word separator - -bash ./tts.sh \ - --stage 1 \ - --lang ar \ -
> --feats_type raw \ - --fs "${fs}" \ - --n_fft "${n_fft}" \ - --n_shift "${n_shift}"
> \ - --token_type char \ - --cleaner tacotron \ - --g2p "${g2p}" \ -
> --train_config "${train_config}" \ - --inference_config "${inference_config}"
> \ - --train_set "${train_set}" \ - --valid_set "${valid_set}" \ -
> --test_sets "${test_sets}" \ - --srctexts "data/${train_set}/text" \ - ${opts}
> "$@" +./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs "${fs}"
> \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ + --token_type char \
> + --cleaner none \ + --g2p none \ + --train_config "${train_config}" \ +
> --inference_config "${inference_config}" \ + --train_set "${train_set}" \
> + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ +
> --srctexts "data/${train_set}/text" \ + ${opts} "$@"
> ------------------------------ In qasr_tts/tts1/README.MD <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873248154>>: > @@
> -0,0 +1,15 @@ +- Our goal is to build a TTS character based system using
> semi-supervised data selection in a low-resource scenario by proposing two
> different methodologies: ⬇️ Suggested change -- Our goal is to build a TTS
> character based system using semi-supervised data selection in a
> low-resource scenario by proposing two different methodologies: +# QASR-TTS
> RECIPE + +- Our goal is to build a TTS character based system using
> semi-supervised data selection in a low-resource scenario by proposing two
> different methodologies: ------------------------------ In
> egs2/TEMPLATE/asr1/db.sh <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873248083>>: > @@
> -1,20 +1,16 @@ # Set the path of your corpus # "downloads" means the corpus
> can be downloaded by the recipe automatically -ACCENTED_FR=downloads No,
> please do not remove other databases from db.sh. You may restore by git
> fetch origin git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh # then
> add your new DB and commit it ... — Reply to this email directly, view it
> on GitHub <#4357 (review)
> <https://github.com/espnet/espnet/pull/4357#pullrequestreview-973301072>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AYBJDB5JUFFZJMEXF5JZL6DVKGE5HANCNFSM5VUH46XA
> . You are receiving this because you were mentioned.Message ID: *@*.***>
>
> Hi, we are not referring to your recipe, but in the case that you remove
> some other entries for other recipes. You may refer the diff here
> https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1132088240>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDBZTHRAZNB576QQPDM3VK2GC3ANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ftshijt commented:
> I added the original db.sh after removing qar_tts since I prevented the automatic download for now.
> […](#)
> On Thu, May 19, 2022, 11:04 PM Jiatong ***@***.***> wrote: I updated the repo. For the db.sh for now we will not add automatic download. Yes, the cleaner is not needed. … <#m_-8517925574695521207_> On Mon, 16 May 2022 at 03:52, Tomoki Hayashi *@*.*> wrote: @.** commented on this pull request. ------------------------------ In qasr_tts/tts1/local/data_download.sh <#4357 (comment) <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873247375)>>: > +# Copyright 2019 Tomoki Hayashi +# Apache 2.0 ( http://www.apache.org/licenses/LICENSE-2.0) Please change to yours. ------------------------------ In qasr_tts/tts1/run.sh <#4357 (comment) <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873247527)>>: > +# g2p=g2p_en # Include word separator +g2p=g2p_en_no_space # Include no word separator + +bash ./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ + --token_type char \ + --cleaner tacotron \ + --g2p "${g2p}" \ + --train_config "${train_config}" \ + --inference_config "${inference_config}" \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts} "$@" G2P and cleaner may not be needed? (Please check) ⬇️ Suggested change -# g2p=g2p_en # Include word separator -g2p=g2p_en_no_space # Include no word separator - -bash ./tts.sh \ - --stage 1 \ - --lang ar \ - --feats_type raw \ - --fs "${fs}" \ - --n_fft "${n_fft}" \ - --n_shift "${n_shift}" \ - --token_type char \ - --cleaner tacotron \ - --g2p "${g2p}" \ - --train_config "${train_config}" \ - --inference_config "${inference_config}" \ - --train_set "${train_set}" \ - --valid_set "${valid_set}" \ - --test_sets "${test_sets}" \ - --srctexts "data/${train_set}/text" \ - ${opts} "$@" +./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ + --token_type char \ + --cleaner none \ + --g2p none \ + --train_config "${train_config}" \ + --inference_config "${inference_config}" \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts} "$@" ------------------------------ In qasr_tts/tts1/README.MD <#4357 (comment) <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873248154)>>: > @@ -0,0 +1,15 @@ +- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies: ⬇️ Suggested change -- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies: +# QASR-TTS RECIPE + +- Our goal is to build a TTS character based system using semi-supervised data selection in a low-resource scenario by proposing two different methodologies: ------------------------------ In egs2/TEMPLATE/asr1/db.sh <#4357 (comment) <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#discussion_r873248083)>>: > @@ -1,20 +1,16 @@ # Set the path of your corpus # "downloads" means the corpus can be downloaded by the recipe automatically -ACCENTED_FR=downloads No, please do not remove other databases from db.sh. You may restore by git fetch origin git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh # then add your new DB and commit it ... — Reply to this email directly, view it on GitHub <#4357 (review) <[#4357 (review)](https://github.com/espnet/espnet/pull/4357#pullrequestreview-973301072)>>, or unsubscribe https://github.com/notifications/unsubscribe-auth/AYBJDB5JUFFZJMEXF5JZL6DVKGE5HANCNFSM5VUH46XA . You are receiving this because you were mentioned.Message ID: *@*.***> Hi, we are not referring to your recipe, but in the case that you remove some other entries for other recipes. You may refer the diff here https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2 — Reply to this email directly, view it on GitHub <[#4357 (comment)](https://github.com/espnet/espnet/pull/4357#issuecomment-1132088240)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYBJDBZTHRAZNB576QQPDM3VK2GC3ANCNFSM5VUH46XA> . You are receiving this because you were mentioned.Message ID: ***@***.***>

But it seems not fix though, are you using the latest db.sh in the master branch?

massabaali7 commented:
I updated it.

On Fri, 20 May 2022 at 20:57, Jiatong ***@***.***> wrote:

> I added the original db.sh after removing qar_tts since I prevented the
> automatic download for now.
> … <#m_7496425465348688300_>
> On Thu, May 19, 2022, 11:04 PM Jiatong *@*.*> wrote: I updated the repo.
> For the db.sh for now we will not add automatic download. Yes, the cleaner
> is not needed. … <#m_-8517925574695521207_> On Mon, 16 May 2022 at 03:52,
> Tomoki Hayashi @.> wrote: @.* commented on this pull request.
> ------------------------------ In qasr_tts/tts1/local/data_download.sh
> <#4357 (comment) <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873247375>>>: >
> +# Copyright 2019 Tomoki Hayashi +# Apache 2.0 (
> http://www.apache.org/licenses/LICENSE-2.0) Please change to yours.
> ------------------------------ In qasr_tts/tts1/run.sh <#4357 (comment) <#4357
> (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873247527>>>: >
> +# g2p=g2p_en # Include word separator +g2p=g2p_en_no_space # Include no
> word separator + +bash ./tts.sh \ + --stage 1 \ + --lang ar \ +
> --feats_type raw \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}"
> \ + --token_type char \ + --cleaner tacotron \ + --g2p "${g2p}" \ +
> --train_config "${train_config}" \ + --inference_config "${inference_config}"
> \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ +
> --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts}
> "$@" G2P and cleaner may not be needed? (Please check) ⬇️ Suggested
> change -# g2p=g2p_en # Include word separator -g2p=g2p_en_no_space #
> Include no word separator - -bash ./tts.sh \ - --stage 1 \ - --lang ar \ -
> --feats_type raw \ - --fs "${fs}" \ - --n_fft "${n_fft}" \ - --n_shift "${n_shift}"
> \ - --token_type char \ - --cleaner tacotron \ - --g2p "${g2p}" \ -
> --train_config "${train_config}" \ - --inference_config "${inference_config}"
> \ - --train_set "${train_set}" \ - --valid_set "${valid_set}" \ -
> --test_sets "${test_sets}" \ - --srctexts "data/${train_set}/text" \ - ${opts}
> "$@" +./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs "${fs}"
> \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ + --token_type char \
> + --cleaner none \ + --g2p none \ + --train_config "${train_config}" \ +
> --inference_config "${inference_config}" \ + --train_set "${train_set}" \
> + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ +
> --srctexts "data/${train_set}/text" \ + ${opts} "$@"
> ------------------------------ In qasr_tts/tts1/README.MD <#4357 (comment) <#4357
> (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873248154>>>: >
> @@ -0,0 +1,15 @@ +- Our goal is to build a TTS character based system using
> semi-supervised data selection in a low-resource scenario by proposing two
> different methodologies: ⬇️ Suggested change -- Our goal is to build a TTS
> character based system using semi-supervised data selection in a
> low-resource scenario by proposing two different methodologies: +# QASR-TTS
> RECIPE + +- Our goal is to build a TTS character based system using
> semi-supervised data selection in a low-resource scenario by proposing two
> different methodologies: ------------------------------ In
> egs2/TEMPLATE/asr1/db.sh <#4357 (comment) <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#discussion_r873248083>>>: >
> @@ -1,20 +1,16 @@ # Set the path of your corpus # "downloads" means the
> corpus can be downloaded by the recipe automatically -ACCENTED_FR=downloads
> No, please do not remove other databases from db.sh. You may restore by git
> fetch origin git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh # then
> add your new DB and commit it ... — Reply to this email directly, view it
> on GitHub <#4357 (review) <#4357 (review)
> <https://github.com/espnet/espnet/pull/4357#pullrequestreview-973301072>>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AYBJDB5JUFFZJMEXF5JZL6DVKGE5HANCNFSM5VUH46XA
> . You are receiving this because you were mentioned.Message ID: *@*.*>
> Hi, we are not referring to your recipe, but in the case that you remove
> some other entries for other recipes. You may refer the diff here
> https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2
> <https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2>
> — Reply to this email directly, view it on GitHub <#4357 (comment)
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1132088240>>, or
> unsubscribe
> https://github.com/notifications/unsubscribe-auth/AYBJDBZTHRAZNB576QQPDM3VK2GC3ANCNFSM5VUH46XA
> <https://github.com/notifications/unsubscribe-auth/AYBJDBZTHRAZNB576QQPDM3VK2GC3ANCNFSM5VUH46XA>
> . You are receiving this because you were mentioned.Message ID: @.*>
>
> But it seems not fix though, are you using the latest db.sh in the master
> branch?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1133122999>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDB2QMSOMT3JKKRJJ5EDVK677LANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

massabaali7 commented:
Hello,
I updated everything but I guess this won't work with this tts.sh bash file
it should work with the one I attached which I am running on my local
machine. Because I ran everything after all the modifications and it's
working.

On Tue, 24 May 2022 at 11:49, Massa Baali ***@***.***> wrote:

> I updated it.
>
> On Fri, 20 May 2022 at 20:57, Jiatong ***@***.***> wrote:
>
>> I added the original db.sh after removing qar_tts since I prevented the
>> automatic download for now.
>> … <#m_-3737783626222192788_m_7496425465348688300_>
>> On Thu, May 19, 2022, 11:04 PM Jiatong *@*.*> wrote: I updated the repo.
>> For the db.sh for now we will not add automatic download. Yes, the cleaner
>> is not needed. … <#m_-8517925574695521207_> On Mon, 16 May 2022 at 03:52,
>> Tomoki Hayashi @.> wrote: @.* commented on this pull request.
>> ------------------------------ In qasr_tts/tts1/local/data_download.sh
>> <#4357 (comment) <#4357 (comment)
>> <https://github.com/espnet/espnet/pull/4357#discussion_r873247375>>>: >
>> +# Copyright 2019 Tomoki Hayashi +# Apache 2.0 (
>> http://www.apache.org/licenses/LICENSE-2.0) Please change to yours.
>> ------------------------------ In qasr_tts/tts1/run.sh <#4357 (comment) <#4357
>> (comment)
>> <https://github.com/espnet/espnet/pull/4357#discussion_r873247527>>>: >
>> +# g2p=g2p_en # Include word separator +g2p=g2p_en_no_space # Include no
>> word separator + +bash ./tts.sh \ + --stage 1 \ + --lang ar \ +
>> --feats_type raw \ + --fs "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}"
>> \ + --token_type char \ + --cleaner tacotron \ + --g2p "${g2p}" \ +
>> --train_config "${train_config}" \ + --inference_config "${inference_config}"
>> \ + --train_set "${train_set}" \ + --valid_set "${valid_set}" \ +
>> --test_sets "${test_sets}" \ + --srctexts "data/${train_set}/text" \ + ${opts}
>> "$@" G2P and cleaner may not be needed? (Please check) ⬇️ Suggested
>> change -# g2p=g2p_en # Include word separator -g2p=g2p_en_no_space #
>> Include no word separator - -bash ./tts.sh \ - --stage 1 \ - --lang ar \ -
>> --feats_type raw \ - --fs "${fs}" \ - --n_fft "${n_fft}" \ - --n_shift "${n_shift}"
>> \ - --token_type char \ - --cleaner tacotron \ - --g2p "${g2p}" \ -
>> --train_config "${train_config}" \ - --inference_config "${inference_config}"
>> \ - --train_set "${train_set}" \ - --valid_set "${valid_set}" \ -
>> --test_sets "${test_sets}" \ - --srctexts "data/${train_set}/text" \ - ${opts}
>> "$@" +./tts.sh \ + --stage 1 \ + --lang ar \ + --feats_type raw \ + --fs
>> "${fs}" \ + --n_fft "${n_fft}" \ + --n_shift "${n_shift}" \ +
>> --token_type char \ + --cleaner none \ + --g2p none \ + --train_config "${train_config}"
>> \ + --inference_config "${inference_config}" \ + --train_set "${train_set}"
>> \ + --valid_set "${valid_set}" \ + --test_sets "${test_sets}" \ +
>> --srctexts "data/${train_set}/text" \ + ${opts} "$@"
>> ------------------------------ In qasr_tts/tts1/README.MD <#4357 (comment) <#4357
>> (comment)
>> <https://github.com/espnet/espnet/pull/4357#discussion_r873248154>>>: >
>> @@ -0,0 +1,15 @@ +- Our goal is to build a TTS character based system using
>> semi-supervised data selection in a low-resource scenario by proposing two
>> different methodologies: ⬇️ Suggested change -- Our goal is to build a TTS
>> character based system using semi-supervised data selection in a
>> low-resource scenario by proposing two different methodologies: +# QASR-TTS
>> RECIPE + +- Our goal is to build a TTS character based system using
>> semi-supervised data selection in a low-resource scenario by proposing two
>> different methodologies: ------------------------------ In
>> egs2/TEMPLATE/asr1/db.sh <#4357 (comment) <#4357 (comment)
>> <https://github.com/espnet/espnet/pull/4357#discussion_r873248083>>>: >
>> @@ -1,20 +1,16 @@ # Set the path of your corpus # "downloads" means the
>> corpus can be downloaded by the recipe automatically -ACCENTED_FR=downloads
>> No, please do not remove other databases from db.sh. You may restore by git
>> fetch origin git checkout origin/master -- egs2/TEMPLATE/asr1/db.sh # then
>> add your new DB and commit it ... — Reply to this email directly, view it
>> on GitHub <#4357 (review) <#4357 (review)
>> <https://github.com/espnet/espnet/pull/4357#pullrequestreview-973301072>>>,
>> or unsubscribe
>> https://github.com/notifications/unsubscribe-auth/AYBJDB5JUFFZJMEXF5JZL6DVKGE5HANCNFSM5VUH46XA
>> . You are receiving this because you were mentioned.Message ID: *@*.*>
>> Hi, we are not referring to your recipe, but in the case that you remove
>> some other entries for other recipes. You may refer the diff here
>> https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2
>> <https://github.com/espnet/espnet/pull/4357/files#diff-78ec6bbf6a24374618d8d7c3bf13181a26d9cd0b714e3014579ccbdd014dabf2>
>> — Reply to this email directly, view it on GitHub <#4357 (comment)
>> <https://github.com/espnet/espnet/pull/4357#issuecomment-1132088240>>, or
>> unsubscribe
>> https://github.com/notifications/unsubscribe-auth/AYBJDBZTHRAZNB576QQPDM3VK2GC3ANCNFSM5VUH46XA
>> <https://github.com/notifications/unsubscribe-auth/AYBJDBZTHRAZNB576QQPDM3VK2GC3ANCNFSM5VUH46XA>
>> . You are receiving this because you were mentioned.Message ID: @.*>
>>
>> But it seems not fix though, are you using the latest db.sh in the master
>> branch?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/espnet/espnet/pull/4357#issuecomment-1133122999>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AYBJDB2QMSOMT3JKKRJJ5EDVK677LANCNFSM5VUH46XA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

kan-bayashi commented on egs2/qasr_tts/tts1/db.sh:
```diff
@@ -0,0 +1 @@
+../../TEMPLATE/asr1/db.sh
```
ditto

ftshijt commented:
A note FYI: you may use `egs2/TEMPLATE/tts1/setup.sh` to automatically create the symlinks

sw005320 commented:
@ftshijt or @kan-bayashi, can you check the PR again?

massabaali7 commented:
I compared with other recipes and ran on my local machine. All is working
fine. What's the problem?

On Wed, Jun 15, 2022, 5:11 PM Tomoki Hayashi ***@***.***>
wrote:

> Files are totally wrong... Why don't you compare with other recipe?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1156453894>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDB3BSDDO7XEHITBI7EDVPHJAHANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

massabaali7 commented:
Not all of them included those files. Okay I will include the setup.sh.

On Wed, Jun 15, 2022, 5:18 PM Tomoki Hayashi ***@***.***>
wrote:

> I compared with other recipes
>
> Really? Your recipe does not include the symlink of
>
>    - path.sh
>    - utils
>    - steps
>    - sids
>    - pyscripts
>    - scripts
>    And it includes wrong symlinks
>    - tts (why no extension...)
>    - setup (setup.sh is used to create recipe template... no need to
>    copy...)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1156462012>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDB4RBT2BCLX7MWLFA63VPHJZRANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

massabaali7 commented:
Okay, I will add all the required files (path.sh, utils, symlinks). I
apologize for the misunderstanding I thought that those files should be
added by the user as I haven't seen them in all of the recipes.

On Wed, Jun 15, 2022, 5:25 PM Shinji Watanabe ***@***.***>
wrote:

> @massabaali7 <https://github.com/massabaali7>, this is an open-source
> activity, and we need to make an example that can be used for everyone.
> So, "ran on my local machine" is not enough.
> All our recipes care about it.
> Please follow @kan-bayashi <https://github.com/kan-bayashi>'s suggestions
> more seriously and check the related documents.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1156470589>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDBY2F2GPDCOZEPXUVVTVPHKUZANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

massabaali7 commented:
I was testing but I pushed it again.

On Tue, 21 Jun 2022 at 15:39, mergify[bot] ***@***.***> wrote:

> This pull request is now in conflict :(
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1161631096>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDBZOVUGKWYNENFAYGFLVQGSX3ANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

massabaali7 commented:
@kan-bayashi @ftshijt do you have any comments on the latest push?

ftshijt commented on egs2/TEMPLATE/asr1/db.sh:
```diff
@@ -1,290 +1 @@
-# Set the path of your corpus
```
You are relinking the template with its own, which cannot work

ftshijt commented on egs2/TEMPLATE/asr1/path.sh:
```diff
@@ -1,22 +1 @@
-MAIN_ROOT=$PWD/../../..
```
Please do not change the template here

ftshijt commented on egs2/TEMPLATE/tts1/tts.sh:
```diff
@@ -1,1235 +1 @@
-#!/usr/bin/env bash
```
Please do not remove the existing template

ftshijt commented on egs2/TEMPLATE/tts1/tts.sh:
```diff
@@ -1,1235 +1 @@
-fi
-
-log "Successfully finished. [elapsed=${SECONDS}s]"
+../../TEMPLATE/tts1/tts.sh
```
It does not make sense to conduct symlink to the same file, while you are deleting it.

ftshijt commented on egs2/qasr_tts/tts1/RESULTS.MD:
```diff
@@ -0,0 +1,32 @@
+
+      - MOS
+
+        - intelligibility: 4.4 ± 0.06
```
It would be preferred if you have more information to how the mos score is collected

ftshijt reviewed (changes requested):
There are still severe issues, please fix them. Also, you did not reflect @kan-bayashi 's issues yet in your code. Please check other recipes or related PR to see what are expected from the recipe-based PR.

ftshijt commented:
FYI: https://github.com/espnet/espnet/pull/4428 This could be an example of what we expected.

massabaali7 commented:
@ftshijt I pushed a new update from ubuntu machine that I borrowed. Pls, let me know if there are still any conflicts. Many thanks for the comments.

massabaali7 commented:
@ftshijt did you still have any comments on the new version? Many thanks, Massa

ftshijt commented:
It's still not fixed yet. You can directly check the file changes of your PR here https://github.com/espnet/espnet/pull/4357/files 

where you can find that you are deleting `asr1/db.sh` `asr1/path.sh` `tts1/tts.sh` and so on

massabaali7 commented:
Thank you for your response. How am I deleting them? I just push the new
commit as is copied from ljspeech with all the required files excluding the
ones I changed. Pls let me know how files are getting deleted should I undo
or should I refork espnet again?

On Fri, Jul 8, 2022, 11:47 PM Jiatong ***@***.***> wrote:

> It's still not fixed yet. You can directly check the file changes of your
> PR here https://github.com/espnet/espnet/pull/4357/files
>
> where you can find that you are deleting asr1/db.sh asr1/path.sh
> tts1/tts.sh and so on
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#issuecomment-1179311708>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDB5ALMIDYELFJY2C6DTVTCAUHANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sw005320 commented:
@massabaali7, frankly, we found a difficulty of working with this PR.
As @kan-bayashi and @ftshijt suggested, we can just follow other PRs and make minimal changes.
Many people can do this without any issue.
However, you do not follow our suggestions and the PR issues remain (e.g., we can still have a conflict and so on.).

I feel that it is difficult for us to fix this issue through this PR.
I recommend you study github PRs and basic open source activities by checking several web resources to be familiar with this kind of activities.
I also recommend you ask your colleagues rather than directly asking us.

ftshijt commented:
I'm closing the PR upon the discussion.

massabaali7 commented:
Ok, I will double check how to submit a recipe and will push it again. So I
can understand how the issues are generated by copying a recipe and by
applying very minor modifications.

Thanks.

On Thu, Aug 4, 2022, 1:17 AM Jiatong ***@***.***> wrote:

> Closed #4357 <https://github.com/espnet/espnet/pull/4357>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/espnet/espnet/pull/4357#event-7123105960>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYBJDBZT4G6X75E5WJKYFO3VXLVYZANCNFSM5VUH46XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
